Add undo feature in Editing plugin

arnaud-morvan opened the pull request.

sbrunner commented on geoext.ux/ux/FeatureEditing/lib/GeoExt.ux/FeatureEditorGrid.js:
```diff
@@ -238,7 +238,7 @@ GeoExt.ux.FeatureEditorGrid = Ext.extend(Ext.grid.EditorGridPanel, {
             ]
         });
-        this.dirty = feature.state == OpenLayers.State.INSERT;
+        this.dirty = !!feature.state;
```
This test looks relay strange for me, Why do you change it ?

sbrunner commented on geoext.ux/ux/FeatureEditing/lib/GeoExt.ux/FeatureEditorGrid.js:
```diff
@@ -309,6 +310,28 @@ GeoExt.ux.FeatureEditorGrid = Ext.extend(Ext.grid.EditorGridPanel, {
         }, this);
+
+        this.undoStack = [];
+        if (!!feature) {
```
How can this test be false ?

sbrunner commented on geoext.ux/ux/FeatureEditing/lib/GeoExt.ux/FeatureEditorGrid.js:
```diff
@@ -318,11 +341,77 @@ GeoExt.ux.FeatureEditorGrid = Ext.extend(Ext.grid.EditorGridPanel, {
+            }
+            else {
+                // Don't stash twice on first modification
+                if (!!e.feature.state) {
```
Also looks strange ...

sbrunner commented on geoext.ux/ux/FeatureEditing/lib/GeoExt.ux/FeatureEditorGrid.js:
```diff
@@ -318,11 +341,77 @@ GeoExt.ux.FeatureEditorGrid = Ext.extend(Ext.grid.EditorGridPanel, {
+     *  Stash feature state and properties for the undo functionality.
+     */
+    stash: function(feature) {
+        if (!!this.previous) {
```
`this.previous !== null` is relay better!, and it should be initialized to to null and documented

sbrunner commented:
I stop my review because I see many boolean operation on non boolean type, this should be avoid ...

arnaud-morvan commented:
I've removed all boolean tests on feature.state, replaced by explicit tests.
I've initialized and documented this.previous and this.undoStack properties.

sbrunner commented:
Excellent, for me it's OK to merge :-)
